undertow/src/main/java/org/wildfly/extension/undertow/ApplicationSecurityDomainDefinition.java

```diff
@@ -354,7 +366,7 @@ public String getSecurityDomainName() {
     }
 
     private static HttpAuthenticationFactory toHttpAuthenticationFactory(final SecurityDomain securityDomain, final String realmName) {
-        final HttpServerAuthenticationMechanismFactory mechanismFactory = new FilterServerMechanismFactory(new ServerMechanismFactoryImpl(), SERVLET_MECHANISM);
+        final HttpServerAuthenticationMechanismFactory mechanismFactory = new FilterServerMechanismFactory(delegate, SERVLET_MECHANISM);
```

Comment:
@parsharma Thanks for working on this! Instead of using `delegate` here, the new construction can be done here.